llvm/lib/Support/Compression.cpp

```diff
@@ -202,6 +207,71 @@ Error zstd::decompress(ArrayRef<uint8_t> Input,
   return E;
 }
 
+static void zstdContextFree(ZSTD_DCtx *Ctx) {
+  size_t RC = ::ZSTD_freeDCtx(Ctx);
+  assert(::ZSTD_isError(RC) == 0);
+}
+
+Error zstd::decompress(ArrayRef<uint8_t> Input,
+                       SmallVectorImpl<uint8_t> &Output) {
+  size_t Res = 0;
+  std::unique_ptr<::ZSTD_DCtx, void (*)(::ZSTD_DCtx *)> Ctx(::ZSTD_createDCtx(),
+                                                            zstdContextFree);
+  Res = ::ZSTD_initDStream(&*Ctx);
+  if (::ZSTD_isError(Res))
+    return make_error<StringError>(::ZSTD_getErrorName(Res),
+                                   inconvertibleErrorCode());
+
+  // Create a buffer, this will be filled up repeatedly by
+  // ZSTD_decompressStream and appended to Output
+  SmallVector<char> Buffer;
+  Buffer.resize_for_overwrite(4096);
+
+  ::ZSTD_inBuffer InputBuffer = {
+      .src = Input.data(), .size = Input.size(), .pos = 0};
+  ::ZSTD_outBuffer OutputBuffer = {
+      .dst = Buffer.data(), .size = Buffer.size(), .pos = 0};
```

Comment:
Is this `clang-format`d?

Reply:
Yes and no, the code I've added **is** clang-formatted, however there are some other parts of the code (that I didn't touch) that change if I run `orc shell clang-format -i Compression.cpp IRReader.cpp`

Reply:
Are you using `clang-format --style=llvm`?

Reply:
No, because there's `llvm/.clang-format` which already specifies it.